core/shared/src/main/scala/org/typelevel/log4cats/AskStructuredLogger.scala

```diff
@@ -0,0 +1,71 @@
+package org.typelevel.log4cats
+
+import cats.Monad
+import cats.mtl.Ask
+
+private[log4cats] class AskStructuredLogger[F[_]](logger: StructuredLogger[F])(implicit
```

Comment:
I know it's out of character for this repo, but could you add some docs and tests?

Reply:
Definitely.  I wanted to iterate on the approach a little bit first before investing too much in that.

This should work for non-slf4j loggers, with the aforementioned performance concerns.  Is it worth trying to do both, or do we think it's better to focus on the slf4j that everyone uses in practice?

Reply:
I don't think the performance can be meaningfully worse than the existing terrible performance of every call to `addContext` adding another `Map` concatenation for *every log operation*